webpack/assets/javascripts/react_app/components/common/Slot/Slot.js

```diff
@@ -0,0 +1,37 @@
+import { cloneElement, isValidElement } from 'react';
+import PropTypes from 'prop-types';
+import {
+  getComponentByWeight,
+  getMaxComponent,
+} from '../Fill/ExtendableRegistery';
+
+const Slot = ({ shouldUpdate, id, multi, children, ...props }) => {
+  const addProps = object => {
+    if (isValidElement(object)) {
+      return cloneElement(object, props);
+    }
+    return cloneElement(children, object);
```

Comment:
shouldn't you keep the old props? `return cloneElement(children, { ...props, ...object });`